ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/HttpJsonConnectorExecutor.java

```diff
@@ -61,8 +61,18 @@ public class HttpJsonConnectorExecutor implements RemoteConnectorExecutor {
     @Getter
     private Client client;
 
+    private CloseableHttpClient httpClient;
+
     public HttpJsonConnectorExecutor(Connector connector) {
+        super.validate(connector);
         this.connector = (HttpConnector) connector;
+        this.httpClient = MLHttpClientFactory
+            .getCloseableHttpClient(super.getConnectionTimeoutInMillis(), super.getReadTimeoutInMillis(), super.getMaxConnections());
```

Comment:
Can we use `getHttpClientConfig` method of Connector class ?

Reply:
Addressed